### PR TITLE
Child Tax Credit (and ODC)

### PR DESCRIPTION
**Pull request by nikhilwoodruff**

### CODE REVIEW

File: openfisca_us/variables/irs/credits/child_tax_credit/refundability.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        "Income Tax liability which the non-refundable CTC can reduce."
+    )
+    definition_period = YEAR
+
+    def formula(tax_unit, period, parameters):
+        tax_liability = tax_unit("income_tax_before_credits", period)
+        credits = add(
+            tax_unit,
+            period,
+            *[
```
Comment (MaxGhenis) [inline openfisca_us/variables/irs/credits/child_tax_credit/refundability.py:19]: name as a constant, remove * once merging my pr

File: openfisca_us/variables/irs/credits/child_tax_credit/refundability.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                "elderly_disabled_credit",
+            ]
+        )
+        return max_(tax_liability - credits, 0)
+
+
+class ctc_child(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "CTC for non-adult dependents"
```
Comment (MaxGhenis) [inline openfisca_us/variables/irs/credits/child_tax_credit/refundability.py:34]:
```suggestion
    label = "CTC for child dependents"
```

File: openfisca_us/variables/irs/credits/child_tax_credit/refundability.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        if ctc.refundable:
+            return entitlement
+        else:
+            return min_(entitlement, reducible_liability)
+
+
+class ctc_adult(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "CTC for children"
```
Comment (MaxGhenis) [inline openfisca_us/variables/irs/credits/child_tax_credit/refundability.py:53]:
```suggestion
    label = "CTC for adult dependents"
```

File: openfisca_us/variables/irs/credits/child_tax_credit/refundability.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    unit = "currency-USD"
+    definition_period = YEAR
+
+    def formula(tax_unit, period, parameters):
+        ctc = parameters(period).irs.credits.child_tax_credit
+        max_amount = tax_unit("ctc_adult_maximum", period)
+        percent_reduction = tax_unit("ctc_percent_reduction", period)
+        reducible_liability = max_(
```
Comment (MaxGhenis) [inline openfisca_us/variables/irs/credits/child_tax_credit/refundability.py:61]: add comment explaining the reduction of ctc_child

File: openfisca_us/variables/irs/credits/child_tax_credit/refundability.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+class nonrefundable_ctc_unclaimable(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Non-refundable CTC unclaimable"
+    unit = "currency-USD"
+    documentation = "Value of CTC not payable due to non-refundability."
+    definition_period = YEAR
+
+    def formula(tax_unit, period, parameters):
+        maximum_ctc = tax_unit("ctc_child_maximum", period) + tax_unit(
```
Comment (MaxGhenis) [inline openfisca_us/variables/irs/credits/child_tax_credit/refundability.py:85]: use add

**Review by MaxGhenis — APPROVED** [review]

**Comment by nikhilwoodruff** [comment]
Finally got the taxcalc structure generation working - 10 unit tests added with random sampling from CPS households with some positive values.

**Pull request merged**